### ISSUE TITLE
Firmware in MKS UTC V1

### ISSUE THREAD

**Issue by acguardia** [makerbase-mks/MKS-THR36-THR42-UTC#3]
I can't upgrade firmware in my MKS UTC V1.
The DFU-Upload-firmware.bat doesn't work
---------------------------------------------------
D:\MKS\CanBus\MKS-THR36-THR42-UTC-main\MKS UTC DFU-Upload\UTC_DFU_Upload>dfu-util.exe -a 0 -s 0x08000000:leave -D klipper.bin -d 0483:df11
dfu-util 0.8

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2014 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to dfu-util@lists.gnumonks.org

Invalid DFU suffix signature
A valid DFU suffix will be required in a future dfu-util release!!!
No DFU capable USB device available

**Comment by sephthir** [comment]
I am also running into this. Running `dfu-util -l` should show the device after plugging it in with the `BOOT` button held, but it returns that no DFU-capable device is present.

**Comment by terraphantm** [comment]
The issue is makerbase configured the boot bits wrong. You can fix it with an STLink. You'll need to set the connection mode to hardware_reset to successfully connect. Then you need to set nBOOT_sel to 0. Afterwards the boot button will work properly.

**Comment by NexGen-3D-Printing** [comment]
> The issue is makerbase configured the boot bits wrong. You can fix it with an STLink. You'll need to set the connection mode to hardware_reset to successfully connect. Then you need to set nBOOT_sel to 0. Afterwards the boot button will work properly.

Incorrect, what you do is put it back in the packet and send the piece of crap back for a refund, it should work out of the box, if they provide updated firmware it should work the way they state, if it doesn't, then its DOA, we should not be beta testers or expected to perform complex tasks like learning how to use an STLink so we can repair their dodgy inept work.

**Comment by HEusebioDH** [comment]
> The issue is makerbase configured the boot bits wrong. You can fix it with an STLink. You'll need to set the connection mode to hardware_reset to successfully connect. Then you need to set nBOOT_sel to 0. Afterwards the boot button will work properly.

@terraphantm 

Can you please describe the connection procedure with the STLink and the change to be made in more detail. I have the same problem.

**Comment by SnakeKVC** [comment]
not active
![изображение](https://github.com/makerbase-mks/MKS-THR36-THR42-UTC/assets/77589815/7e629e48-a513-4e78-ba65-f7540eba705d)

**Comment by SnakeKVC** [comment]
![изображение](https://github.com/makerbase-mks/MKS-THR36-THR42-UTC/assets/77589815/d2652107-f581-4549-a5e7-e3e4da8c1356)
Enable DFU with stm32CubeProgrammer.
Work for me.
Press "Reset" onboard when try connecting.

**Comment by HEusebioDH** [comment]
@SnakeKVC
Thank you very much, I will try again later.
But from what I remember when I tried, it gave an error due to write protection.

**Comment by SnakeKVC** [comment]
And my Klipper configurations
![изображение](https://github.com/makerbase-mks/MKS-THR36-THR42-UTC/assets/77589815/49ccf82c-5767-452d-ae03-53435abf623c)

**Comment by ueatinmycookie** [comment]
Is there a certain driver for windows 10 that I may be missing? I can't connect to the board via stlink and it shows up in device manager under other devices.

**Comment by SnakeKVC** [comment]
> Is there a certain driver for windows 10 that I may be missing? I can't connect to the board via stlink and it shows up in device manager under other devices.

Try zdiag for update or install drivers
https://zadig.akeo.ie/

**Comment by ueatinmycookie** [comment]
Thanks. I still have an issue connecting to stlink. Error: No debug probe detected.

**Comment by SnakeKVC** [comment]
> Thanks. I still have an issue connecting to stlink. Error: No debug probe detected.

have a screenshot with error?
My ST-Link v2
![изображение](https://github.com/makerbase-mks/MKS-THR36-THR42-UTC/assets/77589815/6d222a65-e071-4327-975d-30090eec0f04)

**Comment by ueatinmycookie** [comment]
I have a clone of that one. Are the swd pin numbers on the utc board the same as the stlink?

**Comment by SnakeKVC** [comment]
> I have a clone of that one. Are the swd pin numbers on the utc board the same as the stlink?

Only four wire need
![изображение](https://github.com/makerbase-mks/MKS-THR36-THR42-UTC/assets/77589815/fd38ac1d-b5f4-4256-b4f8-243dbdb2062d)

**Comment by Korgoth1234** [comment]
I tried to do as you described, but I get an error when I try to set nBOOT_SEL to 0:
```
Error: flash loader cannot be loaded.
Error: Downloading Option Bytes Data failed
```
![изображение](https://github.com/makerbase-mks/MKS-THR36-THR42-UTC/assets/16425273/e2f8d527-2898-4b36-a39e-6abfb11a1200)

**Comment by SnakeKVC** [comment]
Wrong checkbox removed
Swap this
![изображение](https://github.com/makerbase-mks/MKS-THR36-THR42-UTC/assets/77589815/452076fb-491a-43a5-9ad3-65566670c392)

**Comment by Korgoth1234** [comment]
> Wrong checkbox removed Swap this ![изображение](https://user-images.githubusercontent.com/77589815/254787112-452076fb-491a-43a5-9ad3-65566670c392.png)
No, I did everything right, you can see that from the log window:
![изображение](https://github.com/makerbase-mks/MKS-THR36-THR42-UTC/assets/16425273/69fe3c12-bcc5-4f15-803a-9dcb83a98171)

**Comment by SnakeKVC** [comment]
Why did you attach the wrong screenshot?

**Comment by Korgoth1234** [comment]
> Why did you attach the wrong screenshot?
So this shows the stm32cube after I clicked "apply".

**Comment by Korgoth1234** [comment]
I recorded a video of my actions:
https://github.com/makerbase-mks/MKS-THR36-THR42-UTC/assets/16425273/66b319ac-8299-4d82-9125-7db35a010223

**Comment by SnakeKVC** [comment]
On my screenshot with chip stm32g0b1 don't have "DUAL_BANK" options.
I don't undestand.

**Comment by ElectroWiz-file** [comment]
با انتخاب نوع ریست مشکل رفع میشه
